# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/html/HTMLSourceElement.cpp
# Commit: 2de3021ee59b
# Full Hash: 2de3021ee59b2fd6be0792548ce10ec45c45ca51
# Author: Boris Chiou <boris.chiou@gmail.com>
# Date: 2022-08-23 02:46:57
# Regressor Bug: 1694741
# File Overlap Count: 1
# Description:
#   Bug 1694741 - Part 2: Factor out the check if the parent is <picture> for <source> and <img>. r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D152708
# ==============================================================================

diff -r 21ad6900a8ad -r 2de3021ee59b dom/html/HTMLSourceElement.cpp
--- a/dom/html/HTMLSourceElement.cpp	Mon Aug 22 20:18:37 2022 +0000
+++ b/dom/html/HTMLSourceElement.cpp	Mon Aug 22 20:18:37 2022 +0000
@@ -94,11 +94,10 @@
   }
   // If we are associated with a <picture> with a valid <img>, notify it of
   // responsive parameter changes
-  Element* parent = nsINode::GetParentElement();
   if (aNameSpaceID == kNameSpaceID_None &&
       (aName == nsGkAtoms::srcset || aName == nsGkAtoms::sizes ||
        aName == nsGkAtoms::media || aName == nsGkAtoms::type) &&
-      parent && parent->IsHTMLElement(nsGkAtoms::picture)) {
+      IsInPicture()) {
     if (aName == nsGkAtoms::media) {
       UpdateMediaList(aValue);
     }